packages/frontend/app/routes/tenants._index.tsx

```diff
@@ -0,0 +1,177 @@
+import { json, type LoaderFunctionArgs } from '@remix-run/node'
+import { useLoaderData, useNavigate } from '@remix-run/react'
+import { Badge, BadgeColor, PageHeader } from '~/components'
+import { Button, Table } from '~/components/ui'
+import { listTenants } from '~/lib/api/tenant.server'
+import { paginationSchema } from '~/lib/validate.server'
+import { checkAuthAndRedirect } from '../lib/kratos_checks.server'
+import { getSession } from '~/lib/session.server'
+
+export const loader = async ({ request }: LoaderFunctionArgs) => {
+  const cookies = request.headers.get('cookie')
+  await checkAuthAndRedirect(request.url, cookies)
+
+  const url = new URL(request.url)
+  const pagination = paginationSchema.safeParse(
+    Object.fromEntries(url.searchParams.entries())
+  )
+
+  if (!pagination.success) {
+    throw json(null, { status: 400, statusText: 'Invalid pagination.' })
+  }
+
+  let isOperator = false
+  const tenants = await listTenants(request, {
+    ...pagination.data
+  })
+
+  let previousPageUrl = '',
+    nextPageUrl = ''
+  if (tenants.pageInfo.hasPreviousPage)
+    previousPageUrl = `/tenants?before=${tenants.pageInfo.startCursor}`
+  if (tenants.pageInfo.hasNextPage)
+    nextPageUrl = `/tenants?after=${tenants.pageInfo.endCursor}`
+
+  let tenantEdges = tenants.edges
+  const tenantPageInfo = tenants.pageInfo
+  if (tenantEdges.length) {
+    const session = await getSession(cookies)
+    const sessionApiSecret = session.get('apiSecret')
+    if (sessionApiSecret && sessionApiSecret.length > 0) {
+      for (const edge of tenantEdges) {
+        const edgeNode = edge.node
+        if (edgeNode && sessionApiSecret === edgeNode.apiSecret) {
+          isOperator = edgeNode.isOperator
+          break
+        }
+      }
+    }
+    tenantEdges = isOperator
+      ? tenants.edges
+      : tenantEdges.filter(({ node }) => node.apiSecret === sessionApiSecret)
```

Comment:
Instead of doing this to determine if the tenant is an operator, we can just specify the `whoami` field in the gql query specified above in the `listTenants` call.  https://github.com/interledger/rafiki/issues/3124#issuecomment-2575672208

This should be used on the /tenant/create page as well. Just a new query for `whoami` and checking if an operator. Currently you can navigate to `/tenant/create` even if you're not an operator.

Reply:
Still possible for me to navigate to `/tenant/create` as a non operator